### PR TITLE
[Vectorized][Feature] support agg function percentile&&percentile_approx

### LINKED ISSUE #8065: [Feature][Vectorized] support  agg function percentile&&percentile_approx
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #8065 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/aggregate_functions/aggregate_function_percentile_approx.h [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    }
+
+    double get() const { return digest->quantile(targetQuantile); }
+
+    void merge(const PercentileApproxState& rhs) {
+        if (init_flag) {
+            DCHECK(digest.get() != nullptr);
+            digest->merge(rhs.digest.get());
+        } else {
+            digest.reset(new TDigest());
```
Comment (BiteTheDDDDt) [inline be/src/vec/aggregate_functions/aggregate_function_percentile_approx.h:70]: here should update inif_flag

File: be/src/vec/aggregate_functions/aggregate_function_percentile_approx.h [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    }
+
+    void reset() {
+        targetQuantile = INIT_QUANTILE;
+        init_flag = false;
+        digest.reset();
+    }
+
+    bool init_flag = false;
+    std::shared_ptr<TDigest> digest;
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_percentile_approx.h:92]: why use `shared_ptr` not `unique_ptr`

File: be/src/vec/aggregate_functions/aggregate_function_percentile_approx.h [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+#include "vec/aggregate_functions/aggregate_function.h"
+#include "vec/columns/columns_number.h"
+#include "vec/data_types/data_type_decimal.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/io/io_helper.h"
+namespace doris::vectorized {
+
+struct PercentileApproxState {
+    static constexpr double INIT_QUANTILE = -1.0;
+    PercentileApproxState() {}
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_percentile_approx.h:32]: = default

File: be/src/vec/aggregate_functions/aggregate_function_percentile_approx.h [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+    void reset() {
+        targetQuantile = INIT_QUANTILE;
+        init_flag = false;
+        digest.reset();
+    }
+
+    bool init_flag = false;
+    std::shared_ptr<TDigest> digest;
+    double targetQuantile = INIT_QUANTILE;
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_percentile_approx.h:93]: `target_quantile`

**Review by HappenLee — COMMENTED** [review]
some problem

**Comment by zhangstar333** [comment]
> some problem

done.

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**